### PR TITLE
feat(INJI-42): Handle database / disk full error

### PR DESCRIPTION
**Pull request by KiruthikaJeyashankar**
Handled "No space left on device" error thrown when the react-native-mmkv-storage or react-native-fs is unable to write the VC data / meta data.

react-native-mmkv-storage or react-native-fs's storage limit is the remaining storage space of the device. When the device's storage limit is exhausted "No space left on device" error is thrown by the library

Issue Link: https://mosip.atlassian.net/browse/INJI-42

### CODE REVIEW

File: components/VcItem.tsx [hunk 1]
```diff
@@ -126,6 +131,18 @@ export const VcItem: React.FC<VcItemProps> = (props) => {
...
   const emptyWalletBindingId = useSelector(service, selectEmptyWalletBindingId);
+  const settingsService = appService.children.get('settings');
+  const storeError = useSelector(service, selectStoreError);
+  const isSavingFailedInIdle = useSelector(service, selectIsSavingFailedInIdle);
+  const vcLabel = useSelector(settingsService, selectVcLabel);
+  const DISMISS = () => service.send(VcItemEvents.DISMISS());
+
+  let storeErrorTranslationPath = 'errors.savingFailed';
+  const isDiskFullError =
+    storeError?.message?.match('No space left on device') != null;
```
Comment (tilak-puli) [inline components/VcItem.tsx:142]: is there any different way to identify the error?

Previous we had to use `SQLITE_FULL` as that was the only way async storage provided. I am wondering if mmkv has a better way? (as message can modified any time in newer release which is a risk)

Comment (KiruthikaJeyashankar) [reply components/VcItem.tsx:142]: This is the message we get when storage is not found in device
`LOG  Message -  write failed: ENOSPC (No space left on device) -> console.log('Message - ', error.message);`

Are you suggesting like replace equality check :  `No space left on device` -> `ENOSPC`?

File: screens/Home/MyVcsTab.tsx [hunk 2]
```diff
@@ -114,6 +122,18 @@ export const MyVcsTab: React.FC<HomeScreenTabProps> = (props) => {
         onDone={controller.ONBOARDING_DONE}
         onAddVc={controller.ADD_VC}
       />
+      <MessageOverlay
```
Comment (tilak-puli) [inline screens/Home/MyVcsTab.tsx:125]: if this code is extracted to a smaller component in VcItem.tsx. You can reuse the same here

File: components/VcItem.tsx (hunk 1 above)
Comment (PuBHARGAVI) [reply components/VcItem.tsx:142]: we have used "No space left on device" since it is more readable compared to ENOSPC

**Comment by tilak-puli** [comment]
Looks fine to me

**Review by tilak-puli — APPROVED** [review]

**Review by vijay151096 — APPROVED** [review]
LGTM.

**Pull request merged**